lib/ngx/process.lua

```diff
@@ -0,0 +1,53 @@
+-- Copyright (C) Yichun Zhang (agentzh)
+
+
+local ffi = require 'ffi'
+local base = require "resty.core.base"
+local errmsg = base.get_errmsg_ptr()
+local FFI_ERROR = base.FFI_ERROR
+local ffi_str = ffi.string
+local ngx_phase = ngx.get_phase
+
+
+local process_type_name = {
+    [0 ]  = "single",
+    [1 ]  = "master",
+    [2 ]  = "signaller",
+    [3 ]  = "worker",
+    [4 ]  = "helper",
+    [99]  = "privileged agent",
+}
+
+
+local C = ffi.C
+local _M = { version = base.version }
+
+
+ffi.cdef[[
+int ngx_http_lua_ffi_enable_privileged_agent(char **err);
+int ngx_http_lua_ffi_get_process_type(void);
+]]
+
+
+function _M.type()
+    local typ = C.ngx_http_lua_ffi_get_process_type()
+    return process_type_name[typ]
+end
+
+
+function _M.enable_privileged_agent()
+    if ngx_phase() ~= "init" then
```

Comment:
It's actually not really about readable, but for portability. We'll soon add a pseudo request to the `init_by_lua*` context (like the `ngx.timer` handler context) to support yield-able Lua API functions like cosockets.